l10n_ec_delivery_note/models/account_edi_document.py

```diff
@@ -0,0 +1,303 @@
+            custom_key = edi_format._get_batch_key(delivery_note, edi_doc.state)
+            key = (edi_format, edi_doc.state, delivery_note.company_id, custom_key)
+            to_process.setdefault(key, self.env["account.edi.document"])
+            to_process[key] |= edi_doc
+
+        delivery_notes = []
+        for key, documents in to_process.items():
+            edi_format, state, company_id, custom_key = key
+            batch = self.env["account.edi.document"]
+            for doc in documents:
+                if edi_format._support_batching(
+                    doc.l10n_ec_delivery_note_id, state=state, company=company_id
+                ):
+                    batch |= doc
+                else:
+                    delivery_notes.append(doc)
+            if batch:
+                delivery_notes.append(batch)
+        return delivery_notes
+
+    @api.model
+    def _process_job(self, documents, doc_type=None):
+        if self.move_id:
+            return super()._process_job(documents, doc_type)
+
+        def _postprocess_post_edi_results(documents, edi_result):
+            attachments_to_unlink = self.env["ir.attachment"]
+            for document in documents:
+                delivery_note = document.l10n_ec_delivery_note_id
+                delivery_note_result = edi_result.get(delivery_note, {})
+                if delivery_note_result.get("attachment"):
+                    old_attachment = document.attachment_id
+                    document.attachment_id = delivery_note_result["attachment"]
+                    if not old_attachment.res_model or not old_attachment.res_id:
+                        attachments_to_unlink |= old_attachment
+                if delivery_note_result.get("success") is True:
+                    document.write(
+                        {
+                            "state": "sent",
+                            "error": False,
+                            "blocking_level": False,
+                        }
+                    )
+                else:
+                    document.write(
+                        {
+                            "error": delivery_note_result.get("error", False),
+                            "blocking_level": delivery_note_result.get(
+                                "blocking_level", DEFAULT_BLOCKING_LEVEL
+                            )
+                            if "error" in delivery_note_result
+                            else False,
+                        }
+                    )
+            attachments_to_unlink.unlink()
+
+        delivery_note = documents.l10n_ec_delivery_note_id
+        documents.edi_format_id.ensure_one()
+        delivery_note.company_id.ensure_one()
+        if len({doc.state for doc in documents}) != 1:
+            raise ValueError(
+                "All account.edi.document of a job should have the same state"
+            )
+        edi_format = documents.edi_format_id
+        state = documents[0].state
```

Comment:
If you are sure if only one, you don't need index

Reply:
Code is take from Odoo server, see https://github.com/odoo/odoo/blob/c95dfd91dfc8646b9309c29715a5f80e362df8a1/addons/account_edi/models/account_edi_document.py#L193

This method is complex to inherit, so copied many code from odoo server but only when `move_id` is empty